### DISCUSSION TITLE
Setting edges and nodes in VueFlow component definition causes edges to underneath workflow

### DISCUSSION THREAD

**Discussion by alexpeverett** [bcakmakoglu/vue-flow#1594] [Q&A] [answered]
Similar issue: https://github.com/bcakmakoglu/vue-flow/discussions/1503

The linked issue above is the same issue we are having again now. Recently we noticed a warning regarding the deprecation of options (nodes and edges) for the useVueFlow method. We refactored the code to use an id, and we inject the nodes and edges in the VueFlow component instead. However, the edges now appear to be underneath the workflow and cannot be clicked on.

Before: 
<img width="293" alt="Screenshot 2024-08-22 at 10 11 57" src="https://github.com/user-attachments/assets/13f923c4-0137-4597-9201-954032c862b7">
<img width="327" alt="Screenshot 2024-08-22 at 10 12 10" src="https://github.com/user-attachments/assets/f6c6f0f2-aecb-48a0-9c63-b4943f5378cc">
<img width="344" alt="Screenshot 2024-08-22 at 10 14 04" src="https://github.com/user-attachments/assets/d6cdb1ab-47b8-4522-8bb5-aa53a44035e6">

After:
<img width="366" alt="Screenshot 2024-08-22 at 10 15 00" src="https://github.com/user-attachments/assets/9eaada83-3adc-4af9-801d-908c6977431c">
<img width="239" alt="Screenshot 2024-08-22 at 10 15 10" src="https://github.com/user-attachments/assets/b98f1211-469b-4fcf-9b1d-b8ffbfe0cf2f">
<img width="506" alt="Screenshot 2024-08-22 at 10 15 33" src="https://github.com/user-attachments/assets/a6fe2df5-4bd6-4451-833a-83f54f4edbff">

Also when applying an id in object format (access state using options API / https://vueflow.dev/guide/vue-flow/state.html#access-state-in-options-api) the warning regarding options persists but the warning doesn't persist when the id is passed as a simple string instead of an object with the string applied to item 'id'.

No warning:
<img width="295" alt="Screenshot 2024-08-22 at 10 17 21" src="https://github.com/user-attachments/assets/56f6012c-2c75-40cf-9e7b-3a59db6fb2a7">

Warning: 
<img width="381" alt="Screenshot 2024-08-22 at 10 17 48" src="https://github.com/user-attachments/assets/e391b47c-1952-4a35-a62c-f1b8d3154083">
<img width="926" alt="Screenshot 2024-08-22 at 10 18 06" src="https://github.com/user-attachments/assets/a26c97e6-8526-4c6a-bc4f-6cb79819bc0f">

**Comment by bcakmakoglu** [comment, marked as answer]
Set `elevate-edges-on-select` on `VueFlow` so that the edges get elevated to the same level as the nodes. Otherwise they are just going to use a z-index that's below nodes by default.

> also when applying an id in object format (access state using options API / [vueflow.dev/guide/vue-flow/state.html#access-state-in-options-api](https://vueflow.dev/guide/vue-flow/state.html#access-state-in-options-api)) the warning regarding options persists but the warning doesn't persist when the id is passed as a simple string instead of an object with the string applied to item 'id'.

That's intentional, stop using an object as an argument for `useVueFlow`.
